HumanExposure/factotum
# Owlfly Sprint (#956)

* Initial search setup on 9400

* Add django objects to returned payload

* Show results in two columns

* Correct label

* Remove raw JSON column; add input box

* Add tests

* Streamline imports

* Display search string in box

* Change elasticsearch port to 9200

* Catching up with Docker config

* Update test for seed data from dev

* Add entire /fontawesome folder to /static

* Replace all "oi" selectors with "fa" selectors

* Remove openiconic files from static folder

* Fix Selenium tests for a completely headless chrome

* Add a convenience manage.py method "loadalldata" for loading all data

* Add Docker files

* Remove old Docker files

* Fix README.md links

* Remove unused services from "elk" composition

* Fix Mac/Windows vs Linux networking differences

* Reconcile overlapping changes from #888 and #872

* Add missing requests library to requirements.txt

* Start adding datagroup picker

* Move variable declarations to environment variables

* Make query string editable when in use, placeholder otherwise

* Tune heap memory for logstash and requirements.txt

* Convert elasticsearch path to settings variable

* Add persistant Kibana optimize volume

Previously, every time Kibana started it would have to run a hefty "optimization" process. This persists it's resulting optimiation from it's inital run. Subsequent startups are significantly faster.

* Add per-datagroup downloading

* Deleting readme after migrating content to wiki

* Change csv download to use a pseudo buffer and StreamingHttpResponse

* remove references to previous view; update test to use streaming_content

* Correct fontawesome path

* Added long example chemical

From http://www.worldwidewords.org/qa/qa-imm1.htm

* 542 qa summary (#913)

* nuke_and_pave command

* QA Summary

* QA Summary tests and teaks

* QA Summary tests and teaks

* updating requirements.txt (#917)

* 896 Link Chemical Page to PUCS (#920)

* Run Beautify on template

* Add href and id to link item

* Add test to existing method

* 905 hotfix test 901 (#934)

* Correct boolean query

* Nurseryweb Spider (#919)

* Initial search setup on 9400

* Add django objects to returned payload

* Show results in two columns

* Correct label

* Remove raw JSON column; add input box

* Add tests

* Streamline imports

* Display search string in box

* Change elasticsearch port to 9200

* Catching up with Docker config

* Update test for seed data from dev

* Add entire /fontawesome folder to /static

* Replace all "oi" selectors with "fa" selectors

* Remove openiconic files from static folder

* Fix Selenium tests for a completely headless chrome

* Add a convenience manage.py method "loadalldata" for loading all data

* Add Docker files

* Remove old Docker files

* Fix README.md links

* Remove unused services from "elk" composition

* Fix Mac/Windows vs Linux networking differences

* Reconcile overlapping changes from #888 and #872

* Add missing requests library to requirements.txt

* Start adding datagroup picker

* Move variable declarations to environment variables

* Make query string editable when in use, placeholder otherwise

* Tune heap memory for logstash and requirements.txt

* Convert elasticsearch path to settings variable

* Add persistant Kibana optimize volume

Previously, every time Kibana started it would have to run a hefty "optimization" process. This persists it's resulting optimiation from it's inital run. Subsequent startups are significantly faster.

* Add per-datagroup downloading

* Deleting readme after migrating content to wiki

* Change csv download to use a pseudo buffer and StreamingHttpResponse

* remove references to previous view; update test to use streaming_content

* Correct fontawesome path

* Added long example chemical

From http://www.worldwidewords.org/qa/qa-imm1.htm

* Flipping logic on datagroup.all_extracted()

* #905 update existing test to assert NOT all_extracted

* #905 make sure the extract form goes away when all_extracted

* Flipping logic on datagroup.all_extracted()

* 914 remove rid (#918)

* Remove unused pre_save code from RawChem model

* Set RID to null when curation is broken

* bug fix for trunctated true_chem values in upload (#924)

* 895 show unique PUCs on chemical page (#921)

* Change query to return unique PUCs

* Add test for unique PUCs

* 848 Fixed issue with ingredient_rank sorting (#933)

* 848 Fixed issue with ingredient_rank sorting

* 848_ingredient_rank_sorting - fixed test and replaced rawchem yaml file

* 848_ingredient_rank_sorting - included migration file

* 825 elastic PUCs (#925)

* Allow Logstash to remember when it was last run

See HumanExposure/factotum_elastic@23a92eed5fa577a79885116236bc8ab9b2c74ba7

* Add new fixtures to test multiple products per document

* Add tests

* Update query to bucket results

* Update template to show both data documents and PUCs

We no longer need to explicitly state which chemicals are found per result now that unique results are bucketed. The template can look more similar to the wireframe found in the comments of #797. The links to PDFs have been removed as they were not in the aforementioned wireframe, however, adding them back should be trivial if desired.

* 879 data document detail redesign (#923)

* #879 overhaul datadocument detail view

* #879 update borders, buttons, scrollspy text, include truechemnam/cas in card

* #879 organize card, add true fileds

* 916 Reported functional use for CPCat records (#922)

* Add report_funcuse to extracted list presence model

Update tests and sample files;

* adding migration to fix merge conflict

* #897 add title to PUC attribute buttons (#949)

* #897 add title to PUC attribute buttons

* #897 add definitions to puctags

* #897 revert puctag yaml

* #897 add definitions

* 898 tag definition (#941)

* #898 add title to tag buttons on product detail page

* #898 dump tags w/ definition

* #898 add test

* #898 revert puctag yaml

* #898 add definitions

* Add fields to page; add seed data values; write test; (#952)

* Move chemical page link from DXTSID to True Chem Name (#953)

* 911 datagroup pagination (#951)

* Flipping logic on datagroup.all_extracted()

* datagroup list pagination

* datagroup list pagination, related refactoring and query optimization

* Update dashboard/tests/functional/test_datagroup_detail.py

Co-Authored-By: Matt Covalt <mcovalt@mailbox.org>

* Update dashboard/views/data_document.py

Co-Authored-By: Matt Covalt <mcovalt@mailbox.org>

* Fix merge conflict

* 867 chemical landing page bubble plot (#950)

* Add tests

* Optimize bubble SQL query

* Add DTXSID lookup for bubble download

* Make bubble plot dependent on size parameters

* Add bubble plot to chemical landing page

* #867 move css into static, change d3 reference

* #867 update js file location

* adding migration to fix merge conflict
